ovsnl/conntrack.go

```diff
@@ -0,0 +1,556 @@
+// Copyright 2017 DigitalOcean.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+// +build linux
+
+package ovsnl
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ti-mo/conntrack"
+	"github.com/ti-mo/netfilter"
+)
+
+//
+// Conntrack aggregator with bounded ingestion + DESTROY aggregation
+// to handle massive bursts of conntrack DESTROY events without OOMing.
+//
+
+// Tunables - adjust for your environment
+const (
+	eventChanSize      = 512 * 1024
+	eventWorkerCount   = 100
+	destroyFlushIntvl  = 100 * time.Millisecond // flush aggregated DESTROYs every 100ms for minimal lag
+	destroyDeltaCap    = 200000                 // maximum distinct (zone,mark) entries in destroyDeltas
+	dropsWarnThreshold = 100                    // threshold of missedEvents to log a stronger warning
+)
+
+// ConntrackEntry represents a single connection tracking entry from the kernel.
+type ConntrackEntry struct {
+	Protocol   string // "tcp", "udp", "icmp" etc.
+	OrigSrc    net.IP
+	OrigDst    net.IP
+	OrigSPort  uint16
+	OrigDPort  uint16
+	ReplySrc   net.IP
+	ReplyDst   net.IP
+	ReplySPort uint16
+	ReplyDPort uint16
+	Zone       uint16
+	Mark       uint32
+	State      string
+}
+
+// ZoneStats holds statistics for a zone
+type ZoneStats struct {
+	TotalCount int
+	Entries    []ConntrackEntry // Only populated if TotalCount > threshold
+}
+
+// ConntrackPerformanceStats represents aggregated performance counters from all CPUs
+type ConntrackPerformanceStats struct {
+	TotalFound         uint32
+	TotalInvalid       uint32
+	TotalIgnore        uint32
+	TotalInsert        uint32
+	TotalInsertFailed  uint32
+	TotalDrop          uint32
+	TotalEarlyDrop     uint32
+	TotalError         uint32
+	TotalSearchRestart uint32
+	CPUs               int
+}
+
+// ConntrackService manages the connection to the kernel's conntrack via Netlink.
+type ConntrackService struct {
+	// No persistent client - connections created as needed
+}
+
+// zmKey is a compact key for (zone,mark)
+type zmKey struct {
+	zone uint16
+	mark uint32
+}
+
+// ZoneMarkAggregator keeps live counts (zone -> mark -> count) with bounded ingestion
+type ZoneMarkAggregator struct {
+	// primary counts (zone -> mark -> count)
+	mu     sync.RWMutex
+	counts map[uint16]map[uint32]int
```

Comment:
nit: comment next to member it discusses please

```suggestion
	// primary counts (zone -> mark -> count)
	counts map[uint16]map[uint32]int
	mu     sync.RWMutex
```